framework/include/base/ExecFlagRegistry.h

```diff
@@ -49,17 +40,17 @@ class ExecFlagRegistry
    */
   const ExecFlagEnum & getDefaultFlags() const { return _default_flags; };
 
-private:
-  ExecFlagRegistry();
-
-  /// So it can be constructed
-  friend ExecFlagRegistry & getExecFlagRegistry();
+  /// Return Singleton instance
+  static ExecFlagRegistry & getExecFlagRegistry();
 
-  /// Mutex for guarding access to _flags
-  std::shared_mutex _flags_mutex;
+  ///@{ Don't allow creation through copy consturction or assignment
+  ExecFlagRegistry(ExecFlagRegistry const &) = delete;
+  void operator=(ExecFlagRegistry const &) = delete;
```

Comment:
also would like to see the move equivalents deleted here

Reply:
Done